packages/js/components/src/experimental-tree-control/tree-item.tsx

```diff
@@ -70,6 +79,15 @@ export const TreeItem = forwardRef( function ForwardedTreeItem(
 							onChange={ ( event ) =>
 								selection.onSelectChild( event.target.checked )
 							}
+							onKeyDown={ ( event ) => {
+								if (
+									event.key === 'Escape' &&
+									props.onEscape
+								) {
+									event.preventDefault();
+									props.onEscape();
+								}
+							} }
```

Comment:
I know it is pretty simple, but I still think it would be good to consolidate the onKeyDown handlers to avoid the duplication of code. I'd pull this out into a function that could be called from both.

Reply:
It actually crossed my mind! Gonna fix this.